tests/integration/test_charm.py

```diff
@@ -68,6 +244,36 @@ async def test_oathkeeper_scale_up(ops_test: OpsTest) -> None:
     assert ops_test.model.applications[APP_NAME].units[1].workload_status == "active"
 
 
+async def test_oathkeeper_scale_down(ops_test: OpsTest) -> None:
+    """Check that oathkeeper works after it is scaled down."""
+    app = ops_test.model.applications[APP_NAME]
+
+    await app.scale(1)
+
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME],
+        status="active",
+        raise_on_blocked=True,
+        timeout=1000,
+    )
+
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
```

Comment:
minor: Have a question about whether we need this line to make sure the app is in active status.

Reply:
`ops_test.model.wait_for_idle` makes sure that the app is active, but not unit(s). This assertion makes sure the unit is active as well.
I added `wait_for_active=True` parameter to `ops_test.model.wait_for_idle`, according to the [docs](https://pythonlibjuju.readthedocs.io/en/latest/api/juju.model.html#juju.model.Model.wait_for_idle) it's an equivalent